src/disputables/index.js

```diff
@@ -0,0 +1,261 @@
+
+  const transactionRequests =
+    (await describeScript(evmScript, apps, org.provider)) || []
+
+  // In order to get the terminal action/s we must search through the `transactionRequests` children (if any)
+  const terminalActions = getTerminalActions(transactionRequests)
+  // In most cases we'll have just a single action to describe
+  // If we do have multiple, we use the first action.
+  const terminalAction = terminalActions[0]
+  const terminalActionAppAddress = terminalAction.to
+
+  // Find the App object corresponding to the app where the terminal action will take place
+  const app = apps.find(app =>
+    addressesEqual(app.address, terminalActionAppAddress)
+  )
+
+  // Get disputed action long description
+  const disputedActionRadspec = terminalActions.length
+    ? buildDisputedActionRadspec(terminalActions)
+    : 'No description'
+
+  // Get disputed action short text
+  // Find method corresponding to the function of the disputed action.
+  const method = await findAppMethodFromIntent(app, terminalAction)
+  const disputedActionText = method ? buildDisputedActionText(app, method) : ''
+
+  return [disputedActionRadspec, disputedActionText]
+}
+
+/**
+ * Gets the array of all children transactions belonging to first step transactions in a forwarding path
+ * @param {Array} transactions Array of transactions corresponding to the first step in a forwarding path
+ * @returns {Array} Array of all childrens.
+ */
+function getTerminalActions(transactions) {
+  return transactions
+    .map(transaction =>
+      transaction.children
+        ? getTerminalActions(transaction.children)
+        : transaction
+    )
+    .flat()
+}
+
+/**
+ * Describes function signature (e.g grantPermission => Grant permission)
+ * @param {String} sig Signature of the function to describe
+ * @returns {String} Described signature
+ */
+function describeFunctionSig(sig) {
+  return sig
+    .replace(/\(.*?\)/, '')
+    .split(/(?=[A-Z])/)
+    .map((s, index) => (index === 0 ? capitalize(s) : s.toLowerCase()))
+    .join(' ')
+}
+
+/**
+ * Builds the disputed action short description
+ * @param {App} app App where the disputed terminal action will take place
+ * @param {Object} method Method corresponding to the disputed terminal action function
+ * @returns {String} Short description of the disputed action
+ */
+function buildDisputedActionText(app, method) {
+  const { artifact, name } = app
+  const role = artifact.roles.find(role => method.roles.includes(role.id))
+
+  // If the terminal action function is not protected by a role, we'll try to describe the function signature
+  // (e.g grantPermission would become Grant permission)
+  const actionText = role?.name || describeFunctionSig(method.sig)
+
+  // If `name` is null most likely means the app is Kernel or ACL
+  const appName = name
+    ? name
+        .split('-')
+        .map(capitalize)
+        .join(' ')
+    : 'System app' // TODO: Update to more meaningfull name?
```

Comment:
Oh interesting, we should have the name available for these system apps!

If they're not available, please add an issue to Connect!

In this case, I think we might want to put something like `Unknown app (<app id>)` instead.

Reply:
Just checked the subgraphs again and i can see that they are available on the Aragon Mainnet subgraph but not on the Rinkeby subgraph. Just notified @0xGabi  about it.